crates/server/src/store/forwarder/authority.rs

```diff
@@ -127,7 +127,11 @@ impl Authority for ForwardAuthority {
         debug_assert!(self.origin.zone_of(name));
 
         debug!("forwarding lookup: {} {}", name, rtype);
-        let name: LowerName = name.clone();
+
+        // Ignore FQDN when we forward DNS queries. Without this we can't look
+        // up addresses from system hosts file.
+        let mut name: Name = name.clone().into();
```

Comment:
This looks like a non-trivial change -- why do you think this is the right thing to do?

Reply:
The query sent to upstream DNS is the same no matter FQDN is enabled or not. It seems that FQDN only affects hosts lookup.

Reply:
Do we want the Forwarder to use the local resolve config at all? Do other DNS servers do that on forwarded requests? I don't think so. I think if we want a Hosts file to be used, we should create a separate Authority type for that...

Reply:
I think `dnsmasq` does that, not too sure though... But maybe we can disable `use_hosts_file` by default in ForwardConfig?

Reply:
That sound good to me.